File: ios/Sources/ACCore/Models/Elements/Accordion.swift
import Foundation

// MARK: - Accordion

public struct Accordion: Codable, Equatable {
    public let type: String = "Accordion"
    public var id: String?
    public var panels: [AccordionPanel]
    public var expandMode: ExpandMode?
    public var spacing: Spacing?
    public var separator: Bool?
    public var height: BlockElementHeight?
    public var isVisible: Bool?
    public var requires: [String: String]?

    public init(
        id: String? = nil,
        panels: [AccordionPanel],
        expandMode: ExpandMode? = nil,
        spacing: Spacing? = nil,
        separator: Bool? = nil,
        height: BlockElementHeight? = nil,
        isVisible: Bool? = nil,
        requires: [String: String]? = nil
    ) {
        self.id = id
        self.panels = panels
        self.expandMode = expandMode
        self.spacing = spacing
        self.separator = separator
        self.height = height
        self.isVisible = isVisible
        self.requires = requires
    }
}

public struct AccordionPanel: Codable, Equatable {
    public var title: String
    public var content: [CardElement]
    public var isExpanded: Bool?

    public init(
        title: String,
        content: [CardElement],
        isExpanded: Bool? = nil
    ) {
        self.title = title
        self.content = content
        self.isExpanded = isExpanded
    }
}

File: android/ac-core/src/main/kotlin/com/microsoft/adaptivecards/core/models/Accordion.kt
package com.microsoft.adaptivecards.core.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.Transient
import kotlinx.serialization.json.JsonElement

@Serializable
@SerialName("Accordion")
data class Accordion(
    @Transient override val type: String = "Accordion",
    override val id: String? = null,
    override val isVisible: Boolean = true,
    override val separator: Boolean = false,
    override val spacing: Spacing? = null,
    override val height: BlockElementHeight? = null,
    override val requires: Map<String, String>? = null,
    override val fallback: JsonElement? = null,
    val panels: List<AccordionPanel>,
    val expandMode: ExpandMode = ExpandMode.SINGLE
) : CardElement

@Serializable
data class AccordionPanel(
    val title: String,
    val content: List<CardElement>,
    val isExpanded: Boolean? = null
)
